src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,624 @@
+```
+
+3. Agglomerate taxonomy at Class rank and save to the `glom_tse` object within `altExp`.
+
+```{r}
+altExp(filtered_tse, "glom_tse") <- agglomerateByRank(filtered_tse, assay.type = "rarefy", rank = "Class", name = "rarefy")
+```
+
+4. Select the top 15 taxa and save them to `rarefy_tse`. We keep our `altExp` object `glom_tse` and `rarefy_tse` separate as it will make things easier later. 
+
+```{r}
+# Get the top features
+top15_rarefy <- getTop(altExp(filtered_tse, "glom_tse"), top = 15L, assay.type = "rarefy", method = "sum")
+
+# Create the new TreeSummarizedExperiment object with assays, colData, and rowData
+rarefy_tse <- altExp(filtered_tse, "glom_tse")[top15_rarefy, ]
+```
+
+## Computing alpha diversity indices
+Alpha diversity is a measure of species diversity in a particular area or an ecosystem. In terms of microbial ecology, analyzing the sample diversity of sequencing data is commonly the first step in assessing differences between microbial environments. Some of the most popular alpha diversity indices reported in microbial community analyses are Chao, abundance-based coverage estimators (ACE), Simpson, and Shannon-Weaver.
+
+Chao1 and ACE are nonparametric estimators that describe diversity as the max number of expected species in the sample (species richness). They consider the proportion of species that have been observed before (“recaptured”) to those that are observed only once. Both Chao1 and ACE underestimate true richness at low sample sizes.
+
+On the other hand, Simpson and Shannon-Weaver, consider relative abundances and depends on both, species richness and the evenness (or equitableness), with which individuals are distributed among the different species. However, both metrics have specific biases. The Shannon-Weaver index places a greater weight on species richness, whereas the Simpson index considers evenness more than richness in its measurement.
+
+More discussion and examples illustrating how sample size and normalization methods can affect alpha diversity metrics in references 4 and 5.
+
+In this example, we're going to calculate the diversity measures for our data with no normalization, our TSS normalized data, and our rarefied data.
+
+1. Now, let's use the `addAlpha` method to compute the alpha diversity indices on our non-normalized data, our rarefied data, and our tss normalized data. We also add a `SampleID` colData column to `filtered_tse` to help with out plotting too. For this example we'll use the `shannon_diversity` index to show alpha diversity, though [see here](https://microbiome.github.io/mia/reference/addAlpha.html) for more index options.
+
+```{r}
+# Add SampleID to colData
+filtered_tse$SampleID <- colnames(filtered_tse)
+
+# Compute alpha diversity on normal counts data
+filtered_tse <- addAlpha(filtered_tse, assay.type = "counts", index = "shannon_diversity", 
+                         name = "shannon_diversity_normal")
```

Comment:
Usually Shannon would be calculated from the full (not filtered) data. Could we do that?